appservice/src/SiteWrapper.ts

```diff
@@ -216,18 +224,18 @@ export class SiteWrapper {
         });
 
         const requestApi: request.RequestAPI<request.Request, request.CoreOptions, {}> = request.defaults(httpRequest);
-        const disposable: vscode.Disposable = { dispose: undefined };
+        const logStream: ILogStream = { dispose: undefined, isConnected: true };
         // Intentionally setting up a separate telemetry event and not awaiting the result here since log stream is a long-running action
         // tslint:disable-next-line:no-floating-promises
         actionHandler.callWithTelemetry('appService.streamingLogs', async () => {
             await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
                 const logsRequest: request.Request = requestApi(`${this.kuduUrl}/api/logstream/${path}`);
-                const disconnectMessage: string = localize('logStreamDisconnected', 'Disconnected from log-streaming service.');
-                disposable.dispose = (): void => {
+                logStream.dispose = (): void => {
                     logsRequest.removeAllListeners();
                     logsRequest.destroy();
                     outputChannel.show();
-                    outputChannel.appendLine(disconnectMessage);
+                    outputChannel.appendLine(localize('logStreamDisconnected', 'Disconnected from log-streaming service.'));
```

Comment:
Was this out of scope in the last PR?  Just wondering why you moved it

Reply:
`disconnectMessage` was used twice in the previous PR (once in `dispose` and once in `on('complete'`). I changed it so that `on('complete'` calls `dispose` and now the variable is only used once (aka no reason for a separate variable)